lib/fortitude/widget/caching.rb

```diff
@@ -0,0 +1,38 @@
+module Fortitude
+  class Widget
+    module Caching
+      extend ActiveSupport::Concern
+      include ActionView::Helpers::CacheHelper
+
+      module ClassMethods
+        # PUBLIC API
+        def cacheable(opts = {})
+          if extra_assigns == :use
+            extra_assigns :ignore
+          end
+
+          static_keys = Array(opts.fetch(:key, []))
+          options = opts.fetch(:options, {})
+
+          define_method(:cache_contents) do |&block|
+            cache calculate_cache_dependencies(assigns, static_keys), options do
+              block.call
+            end
+          end
+
+          around_content :cache_contents
+        end
+      end
+
+      private
+
+      def calculate_cache_dependencies(assigns, static_keys)
+        (
+          assigns.to_a.sort_by(&:first).flatten +
```

Comment:
This is me trying to figure out the best way to generate a cache key from the `assigns` hash. By coercing to an array (instead of just using the `#values`), we avoid key collisions like this:

```rb
{ foo: @bar, baz: nil }
```

vs

```rb
{ foo: nil, baz: @bar }
```

To be honest, I'm not really sure why I'm sorting the array. I guess I'm not sure if I'll receive `assigns` in the same order every time, and I want to maximize the potential of a cache hit.